lib/loggly/common.js

```diff
@@ -165,41 +156,67 @@ common.loggly = function () {
   if (auth) {
     requestOptions.headers.authorization = 'Basic ' + new Buffer(auth.username + ':' + auth.password).toString('base64');
   }
-  if (requestBody) {
+  function popMsgsAndSend() {
+    if (isBulk) {
+      var bulk = createBulk(arrMsg);
+      sendBulkLogs(bulk);
+    } else {
+      var input = createInput(requestBody);
+      sendInputLogs(input);
+    }
+  }
+  function createBulk(msgs) {
+    var bulkMsg = {};
+    bulkMsg.msgs = arrMsg.slice();
+    bulkMsg.attemptNumber = 1;
+    bulkMsg.sleepUntilNextRetry = 2 * 1000;
+    bulkMsg.id = bulkId++;
+
+    return bulkMsg;
+  }
+  function createInput(msgs) {
+    var inputMsg = {};
+    inputMsg.msgs = requestBody;
     requestOptions.body = requestBody;
-    arrRetryLogs = arrRetryLogs.concat(requestBody);
+    inputMsg.attemptNumber = 1;
+    inputMsg.sleepUntilNextRetry = 2 * 1000;
+    inputMsg.id = inputId++;
+
+    return inputMsg;
   }
-  function sendLogs() {
-    if (arrRetryLogs.length && !requestBody) requestOptions.body = arrRetryLogs[0];
+  function sendInputLogs(input) {
     try {
       request(requestOptions, function (err, res, body) {
-        if (err) return onError(err);
-        statusCode = res.statusCode.toString();
-        if(statusCode === '403') isValidToken = false;
-        if (statusCode === '500' || statusCode === '503' || statusCode === '504') retryOnServerError(res);
-        if (statusCode === '200') {
-          arrRetryLogs.splice(0, 1);
-          totalRetries = 0;
-        }
-        if (Object.keys(failCodes).indexOf(statusCode) !== -1) {
-          if (statusCode !== '503' && statusCode !== '500' && statusCode !== '504') {
-            return onError((new Error('Loggly Error (' + statusCode + '): ' + failCodes[statusCode])));
+        if (err) {
+          // In rare cases server is busy
+          if (err.code === 'ETIMEDOUT' || err.code === 'ECONNRESET' || err.code === 'ESOCKETTIMEDOUT' || err.code === 'ECONNABORTED') {
+            retryOnError(input, err);
+          } else {
+            return onError(err);
+          }
+        } else {
+            responseCode = res.statusCode;
+            if (responseCode === httpStatusCode.badToken.code) {
+              isValidToken = false;
+              return onError((new Error('Loggly Error (' + responseCode + '): ' + httpStatusCode.badToken.message)));
+            }
+            if (responseCode === httpStatusCode.success.code && input.attemptNumber >= eventRetried) {
```

Comment:
@mchaudhary This condition checks if the event sent successfully after retrying "n" number of times with status code 200 then print a console message.